### PR TITLE
feat(phone): Add creat methods for recovery phone lib

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We want to be able to create recovery phone entries for a user

## This pull request

- Adds methods to create a recovery phone
- Adds database migration to create the recovery phone table

## Issue that this pull request solves

Closes: https://mozilla-hub.atlassian.net/browse/FXA-10341
Closes: https://mozilla-hub.atlassian.net/browse/FXA-10330

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: libs/accounts/recovery-phone/src/lib/recovery-phone.errors.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { BaseError } from '@fxa/shared/error';
+
+export class RecoveryPhoneError extends BaseError {
```
Comment (dschom) [inline libs/accounts/recovery-phone/src/lib/recovery-phone.errors.ts:7]: Nice!

File: libs/accounts/recovery-phone/src/lib/recovery-phone.manager.in.spec.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          useValue: db,
+        },
+      ],
+    }).compile();
+
+    recoveryPhoneManager = moduleRef.get(RecoveryPhoneManager);
+  });
+
+  afterEach(async () => {
+    await db.destroy();
```
Comment (dschom) [inline libs/accounts/recovery-phone/src/lib/recovery-phone.manager.in.spec.ts:30]: Do we need to create and destroy the db after each test, or can we just do this once for the test suite.

File: libs/accounts/recovery-phone/src/lib/recovery-phone.manager.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+   * @param phoneNumber Phone number in E.164 format.
+   */
+  async registerPhoneNumber(uid: string, phoneNumber: string): Promise<any> {
+    if (!this.isE164Format(phoneNumber)) {
+      throw new RecoveryNumberInvalidFormatError(uid, phoneNumber);
+    }
+
+    const uidBuffer = Buffer.from(uid, 'hex');
+
+    // TODO: Perform phone number validation here via https://www.twilio.com/docs/lookup/v2-api#making-a-request
```
Comment (dschom) [inline libs/accounts/recovery-phone/src/lib/recovery-phone.manager.ts:41]: Is there a follow up for this?

File: libs/accounts/recovery-phone/src/lib/recovery-phone.types.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+export interface RecoveryPhone {
+  uid: Buffer;
+  phoneNumber: string;
+  createdAt: number;
+  lastConfirmed: number;
+  lookupData: string;
```
Comment (dschom) [inline libs/accounts/recovery-phone/src/lib/recovery-phone.types.ts:10]: What is lookup data?

File: packages/db-migrations/databases/fxa/patches/patch-158-159.sql [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+-- `lookupData` contains data from https://www.twilio.com/docs/lookup/v2-api
+CREATE TABLE recoveryPhones (
+    uid BINARY(16) NOT NULL,
+    phoneNumber VARCHAR(15) NOT NULL,
+    createdAt BIGINT UNSIGNED NOT NULL,
+    lastConfirmed BIGINT UNSIGNED NOT NULL,
+    lookupData JSON,
+    PRIMARY KEY (uid),
+    INDEX idx_phoneNumber (phoneNumber),
+    FOREIGN KEY (uid) REFERENCES accounts(uid) ON DELETE CASCADE
```
Comment (dschom) [inline packages/db-migrations/databases/fxa/patches/patch-158-159.sql:17]: Nice

**Review by dschom — APPROVED** [review]
Looks good to me!

File: libs/accounts/recovery-phone/src/lib/recovery-phone.manager.ts (hunk 3 above)
Comment (vbudhram) [reply libs/accounts/recovery-phone/src/lib/recovery-phone.manager.ts:41]: Filed https://mozilla-hub.atlassian.net/browse/FXA-10727

File: libs/accounts/recovery-phone/src/lib/recovery-phone.types.ts (hunk 4 above)
Comment (vbudhram) [reply libs/accounts/recovery-phone/src/lib/recovery-phone.types.ts:10]: Looks like it information if the phone number is fradulent or active and can recieve message.

**Pull request merged**